src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -81,21 +91,25 @@ private static boolean isValidPath(Path filePath) {
     /**
      * Saves all data to this storage file.
      *
-     * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * @throws StorageOperationException
+     *             if there were errors converting and/or storing data to file.
      */
     public void save(AddressBook addressBook) throws StorageOperationException {
 
-        /* Note: Note the 'try with resource' statement below.
-         * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
+        /*
+         * Note: Note the 'try with resource' statement below. More info:
+         * https://docs.oracle.com/javase/tutorial/essential/exceptions/
+         * tryResourceClose.html
          */
-        try (final Writer fileWriter =
-                     new BufferedWriter(new FileWriter(path.toFile()))) {
+        try (final Writer fileWriter = new BufferedWriter(new FileWriter(path.toFile()))) {
 
             final AdaptedAddressBook toSave = new AdaptedAddressBook(addressBook);
             final Marshaller marshaller = jaxbContext.createMarshaller();
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
 
+        } catch (FileDeletedException fde) {
```

Comment:
you should be doing this in the `save()` method. try recollecting the discussion around this during the tutorial